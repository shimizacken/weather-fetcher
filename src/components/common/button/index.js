export { Button } from './Button';
export { ButtonTypes } from './buttonTypes';
