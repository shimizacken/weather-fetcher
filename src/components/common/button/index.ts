export { Button } from './Button';
export { ButtonType } from './buttonType';
export { BackButton } from './BackButton';
