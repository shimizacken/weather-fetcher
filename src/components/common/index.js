export { Button } from './button/Button';
export { SearchBox } from './searchBox/SearchBox';
export { RadioButton } from './radioButton/RadioButton';
export { Loader } from './loader/Loader';
export { Row } from './table';
export { Modal, ConfirmationModal } from './modal';
