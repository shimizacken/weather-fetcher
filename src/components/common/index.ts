export {Button} from './button/Button';
export {ButtonType} from './button/buttonType';
export {SearchBox} from './searchBox/SearchBox';
export {RadioButton} from './radioButton/RadioButton';
export {Loader} from './loader/Loader';
export {Table, Row, Column} from './table';
export {Modal, ConfirmationModal} from './modal';
export {BackButton} from './button';
export {TextInput} from './textInput/TextInput';
export {Logo} from './logo/Logo';
