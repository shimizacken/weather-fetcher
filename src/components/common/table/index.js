export { Row } from './row/Row';
