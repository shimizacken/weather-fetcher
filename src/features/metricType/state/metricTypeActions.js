import { SET_TEMPERATURE_UNIT } from './constants';

export const setTempUnit = (tempType) => ({
  type: SET_TEMPERATURE_UNIT,
  tempType
});
