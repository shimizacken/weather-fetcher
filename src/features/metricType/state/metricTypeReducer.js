import { Units } from 'app/services/openWeatherMap/units';
import { SET_TEMPERATURE_UNIT } from './constants';

const initialState = Units.celsius;

export const metricType = (state = initialState, action) => {
  if (action.type === SET_TEMPERATURE_UNIT) {
    return action.tempType;
  }

  return state;
};
