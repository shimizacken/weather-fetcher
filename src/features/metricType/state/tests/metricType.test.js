/* eslint-disable no-undef */
import {randomAction} from 'app/state/common/randomAction';
import {metricType} from '../metricTypeReducer';
import {setTempUnit} from '../metricTypeActions';
import {Units} from 'app/services/openWeatherMap';

describe('metricType test suite', () => {
    describe('default', () => {
        it('should return celsius type (metric) as initialState, when not match action is passed', () => {
            const action = randomAction();

            expect(metricType(undefined, action)).toMatchSnapshot();
        });
    });

    describe('default', () => {
        it('should return celsius type (metric)', () => {
            const state = {
                metricType: undefined,
            };
            const action = setTempUnit(Units.celsius);

            expect(metricType(state, action)).toMatchSnapshot();
        });

        it('should return fahrenheit type (imperial)', () => {
            const state = {
                metricType: undefined,
            };
            const action = setTempUnit(Units.fahrenheit);

            expect(metricType(state, action)).toMatchSnapshot();
        });

        it('should return kelvin type (empty string)', () => {
            const state = {
                metricType: undefined,
            };
            const action = setTempUnit(Units.kelvin);

            expect(metricType(state, action)).toMatchSnapshot();
        });
    });
});
