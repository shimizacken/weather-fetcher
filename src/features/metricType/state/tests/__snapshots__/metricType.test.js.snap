// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`metricType test suite default should return celsius type (metric) 1`] = `
Object {
  "name": "celsius",
  "symbol": "°C",
  "value": "metric",
}
`;

exports[`metricType test suite default should return celsius type (metric) as initialState, when not match action is passed 1`] = `
Object {
  "name": "celsius",
  "symbol": "°C",
  "value": "metric",
}
`;

exports[`metricType test suite default should return fahrenheit type (imperial) 1`] = `
Object {
  "name": "fahrenheit",
  "symbol": "°F",
  "value": "imperial",
}
`;

exports[`metricType test suite default should return kelvin type (empty string) 1`] = `
Object {
  "name": "kelvin",
  "symbol": "K",
  "value": "",
}
`;
